dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/EventExporterTest.java

```diff
@@ -95,4 +115,207 @@ void testExportEventsWhenFilteringByEnrollment()
+                null, null ) ) );
+
+        Events events = eventService.getEvents( params );
+        assertNotNull( events );
+        assertEquals( 1, events.getEvents().size() );
+    }
+
+    @Test
+    void testExportEventsWhenFilteringByDataElementsWithStatusFilter()
+    {
+        EventSearchParams params = new EventSearchParams();
+        params.setOrgUnit( orgUnit );
+        params.setProgramInstances( Sets.newHashSet( "TvctPPhpD8u" ) );
+        params.setProgramStatus( ProgramStatus.ACTIVE );
+
+        DataElement dataElement = dataElement( "DATAEL00001" );
+
+        params.setDataElements( Set.of(
+            new QueryItem( dataElement, QueryOperator.LIKE, "val", dataElement.getValueType(),
+                null, null ) ) );
+
+        Events events = eventService.getEvents( params );
+        assertNotNull( events );
+        assertEquals( 1, events.getEvents().size() );
+    }
+
+    @Test
+    void testExportEventsWhenFilteringByDataElementsWithProgramTypeFilter()
+    {
+        EventSearchParams params = new EventSearchParams();
+        params.setOrgUnit( orgUnit );
+        params.setProgramInstances( Sets.newHashSet( "TvctPPhpD8u" ) );
+        params.setProgramType( ProgramType.WITH_REGISTRATION );
+
+        DataElement dataElement = dataElement( "DATAEL00001" );
+
+        params.setDataElements( Set.of(
+            new QueryItem( dataElement, QueryOperator.LIKE, "val", dataElement.getValueType(),
+                null, null ) ) );
+
+        Events events = eventService.getEvents( params );
+        assertNotNull( events );
+        assertEquals( 1, events.getEvents().size() );
+    }
+
+    @Test
+    void testExportEventsWhenFilteringByDataElementsEqual()
+    {
+        EventSearchParams params = new EventSearchParams();
+        params.setOrgUnit( orgUnit );
+        params.setProgramInstances( Sets.newHashSet( "TvctPPhpD8u" ) );
+
+        DataElement dataElement = dataElement( "DATAEL00001" );
+
+        params.setDataElements( Set.of(
+            new QueryItem( dataElement, QueryOperator.EQ, "value00001", dataElement.getValueType(),
+                null,
+                null ) ) );
+
+        Events events = eventService.getEvents( params );
+        assertNotNull( events );
+        assertEquals( 1, events.getEvents().size() );
+    }
+
+    @Test
+    void testExportEventsWhenFilteringByDataElementsIn()
+    {
+        EventSearchParams params = new EventSearchParams();
+        params.setOrgUnit( orgUnit );
+        params.setProgramInstances( Sets.newHashSet( "TvctPPhpD8u", "TvctPPhpD8z" ) );
+
+        DataElement datael00001 = dataElement( "DATAEL00001" );
+
+        params.setDataElements( Set.of(
+            new QueryItem( datael00001, QueryOperator.IN, "value00001;value00002", datael00001.getValueType(),
+                null,
+                null ) ) );
+
+        Events events = eventService.getEvents( params );
+        assertNotNull( events );
```

Comment:
We are regularly having issues with tests failing on master only on CI with little to no context.

We should absolutely avoid assertions like this for this reason. Imagine this test fails on GitHub only telling you `expected 1 got 2`. What are you going to do? 😅 Multiple devs lost a few days on multiple occasions trying to debug tests with such assertions 😬 

Please use for example
https://junit.org/junit5/docs/5.8.2/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertAll(java.lang.String,java.util.Collection)

like in https://github.com/dhis2/dhis2-core/blob/0d367e9b804ae5476694ab9545840e21b4cd68e5/dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/trigramsummary/TrigramSummaryControllerTest.java#L243-L245

to group the assertion for checking the size and the content below into one assertAll